nemo/collections/nlp/models/dialogue/intent_bio_type_classification_model.py

```diff
@@ -0,0 +1,793 @@
+        logging.info(f'Setting data_dir to {data_dir}.')
+        self.data_dir = data_dir
+
+    @typecheck()
+    def forward(self, input_ids, attention_mask, token_type_ids, mention_mask):
+        """
+        No special modification required for Lightning, define it as you normally would
+        in the `nn.Module` in vanilla PyTorch.
+        """
+        if self._cfg.tokenizer.get('library', '') == 'megatron':
+            hidden_states, _ = self.bert_model(input_ids, attention_mask, tokentype_ids=token_type_ids, lm_labels=None)
+        else:
+            hidden_states = self.bert_model(
+                input_ids=input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
+            )
+
+        from os.path import exists
+        global UNIQUE_EPOCH_COUNTER
+
+        # print("START DUMP hidden_states")
+        # path_to_file="/home/lilee/pickle/hidden_states"+str(UNIQUE_EPOCH_COUNTER)+".pickle"
+        # if exists(path_to_file)==False:
+        #     with open(path_to_file, 'wb') as f:
+        #         # Pickle the 'data' dictionary using the highest protocol available.
+        #         pickle.dump(hidden_states, f, pickle.HIGHEST_PROTOCOL)        
+        # print("END DUMP hidden_states")
+
+
+        # intent_logits, slot_logits = self.classifier(hidden_states=hidden_states)
+        intent_logits, bio_slot_logits = self.classifier(hidden_states=hidden_states)
+
+        # print("START DUMP bio_slot_logits")
+        # path_to_file="/home/lilee/pickle/bio_slot_logits"+str(UNIQUE_EPOCH_COUNTER)+".pickle"
+        # if exists(path_to_file)==False:
+        #     with open(path_to_file, 'wb') as f:
+        #         # Pickle the 'data' dictionary using the highest protocol available.
+        #         pickle.dump(bio_slot_logits, f, pickle.HIGHEST_PROTOCOL)
+        # print("END DUMP bio_slot_logits")
+
+        # current:bug; hidden_states#(batch*128*768) zzz
+        hidden_states_dim = hidden_states.size()[2]
+        max_token_len = hidden_states.size()[1]
+
+        mention_hidden_states=[]
+
+        for m, h in zip(mention_mask, hidden_states):
+            mention_hidden_states_temp=[]
+            m = m.tolist()
+            h = h.tolist()
+            
+            for idx, l in enumerate(m):
+                if l==1:
+                    one_mention_hidden_states=[]
+                    one_mention_hidden_states.append(h[idx])
+                    idx+=1
+                    while(idx<len(m) and m[idx]==2):
+                        one_mention_hidden_states.append(h[idx])
+                        idx+=1
+                    idx-=1
+                    one_mention_hidden_states = torch.FloatTensor(one_mention_hidden_states)
+                    one_mention_mean_emb = torch.mean(one_mention_hidden_states, 0)
+                    mention_hidden_states_temp.append(one_mention_mean_emb.tolist())
+            mention_hidden_states.append(mention_hidden_states_temp)
+
+
+        mention_hidden_states_pad=[]
```

Comment:
I see you're padding to max token len (something like 128) - this is fine for now but before deployment, let's try to reduce to max num entities (something like 5)